### DISCUSSION TITLE
Opayo (Ebizmarts) SagePaySuite incompatibility with OpenMage-LTS 19.x in phpseclib

### DISCUSSION THREAD

**Discussion by durzel** [OpenMage/magento-lts#4053] [Extensions & Modules]
Hi

I've got a Magento 1 CE 1.9.4.5 website that has OpenMage LTS 19.5.3 installed upon it.  I am also using Opayo (Ebizmarts) SagePaySuite 10.0.1 - latest at time of writing.

When OpenMage is installed, attempting to click on any customer with this extension installed produces an error, and the page fails to load.

The error is as follows:

```
2024-06-25T14:04:05+00:00 ERR (3):
Error: Class 'phpseclib\Crypt\Base' not found in /home/magento/public_html/lib/phpseclib/Crypt/Rijndael.php:64
Stack trace:
#0 /home/magento/.modman/sage-pay-suite-pro-10.0.1/app/code/local/Ebizmarts/SagePaySuite/Model/SagePayForm.php(4): include_once()
#1 /home/magento/public_html/lib/Varien/Autoload.php(58): include('/home/magento/....')
#2 [internal function]: Varien_Autoload->autoload()
#3 [internal function]: spl_autoload_call()
....
```

The line in question in `SagePayForm.php` (marked in bold) on line 4 is:
<pre>
use phpseclib\Crypt\Rijndael;
<b>include_once BP . DS . "lib/phpseclib/Crypt/Rijndael.php";</b>
/**
 * FORM main model
</pre>

Line 64 in `/home/magento/public_html/lib/phpseclib/Crypt/Rijndael.php` is (marked in bold):
<pre>
 * @author  Jim Wigginton <terrafrost@php.net>
 * @access  public
 */
<b>class Rijndael extends Base</b>
{
    /**
     * The mcrypt specific name
</pre>

If I comment out this line, the customer pages load, but I don't know if this module fails elsewhere.  Curiously, there is no differences between `lib/phpseclib/Crypt/Rijndael.php` between Magento CE 1.9.4.5 and OpenMage LTS 19.5.3 so I'm a bit of a loss to explain why attempting to look at a customer record works on the former but not the latter, with no other changes.

I have tried speaking to Ebizmarts but they only confirm compatibility with Magento 1 CE, not OpenMage, and since OpenMage breaks it I was hoping someone might be able to offer some insight.

Thanks in advance!

**Comment by kiatng** [comment]
There are many ways to install/upograde OpenMage over Magento 1. How did you install it?

I know nothing about phpseclib. Have you check the [source at GitHub](https://github.com/phpseclib/phpseclib/tree/2.0)? It seems there is newer version explained [here](https://phpseclib.com/docs/why). It's likely that you are using the older version 2. Depending on how you install OM, you may need to reinstall phpseclib with `composer require phpseclib/phpseclib:~2.0`.

**Reply by durzel** [reply]
Thanks.

I installed it via composer, following the instructions here: https://www.openmage.org/magento-lts/install.html

The composer.lock file that the install process creates has this in, for what it's worth:

<pre>
        {
            "name": "phpseclib/phpseclib",
            "version": "3.0.38",
            "source": {
                "type": "git",
                "url": https://github.com/phpseclib/phpseclib.git,
                "reference": "b18b8788e51156c4dd97b7f220a31149a0052067"
            },
</pre>

**Reply by kiatng** [reply]
If installed with composer, then comment out line 4 ```include_once BP . DS . "lib/phpseclib/Crypt/Rijndael.php";``` should be correct. This is because ```Mage``` autoloads the composer vendor.
